code/modules/modular_computers/networking/device_types/_network_device.dm

```diff
@@ -196,6 +226,181 @@
+		LAZYSET(command_and_write, new_alias, LAZYACCESS(command_and_write, old_alias))
+		LAZYREMOVE(command_and_write, old_alias)
+		return TRUE
+
+/datum/extension/network_device/proc/remove_command(alias)
+	if(LAZYACCESS(command_and_call, alias))
+		LAZYREMOVE(command_and_call, alias)
+		return TRUE
+	if(LAZYACCESS(command_and_write, alias))
+		LAZYREMOVE(command_and_write, alias)
+		return TRUE
+	return FALSE
+
+// Any modification of public vars and calling methods from network commands. Return text feedback on success/unsuccess of command.
+/datum/extension/network_device/proc/on_command(command, command_args, mob/user)
+	var/datum/command_target = get_command_target()
+	if(!has_commands)
+		return "Device cannot receive commands."
+	if(!command_target)
+		return "No valid target found for command '[command]'"
+	if(!has_access(user))
+		return "Access denied."
+	if(command_and_call[command])
+		var/decl/public_access/public_method/method = command_and_call[command]
+		method.perform(command_target, command_args)
+		return "Successfully called method '[command]' on [network_tag]."
+	if(command_and_write[command])
+		var/decl/public_access/public_variable/variable = command_and_write[command]
+		if(command_args) // Write to a var.
+			command_args = sanitize_command_args(command_args, variable.var_type)
+			if(isnull(command_args)) // Command could not be converted into a form appropriate for the variable.
+				return "Improper argument type; correct type is [variable.var_type]"
+			if(variable.write_var_protected(command_target, command_args))
+				return "Successfully set variable '[command]' on [network_tag] to [command_args]."
+			return "Unable to set variable '[command]' on [network_tag]."
+		else			 // Read from a var.
+			return "Variable '[command]' on [network_tag] has value: [variable.access_var(command_target)]"
+	return "Command '[command]' not found on [network_tag]."
+
+// Returns the variable converted into the proper form, or null if it is unable to.
+/datum/extension/network_device/proc/sanitize_command_args(command_args, var_type)
+	// First check if the command is a list; if it is, only accept it if the expected type is a list.
+	if(islist(command_args))
+		if(var_type == IC_FORMAT_LIST)
+			return command_args
+		else
+			return null
+	switch(var_type)
+		if(IC_FORMAT_ANY)
+			return command_args
+		if(IC_FORMAT_STRING)
+			return "[command_args]"
+		if(IC_FORMAT_CHAR)
+			if(istext(command_args) && length(command_args) == 1)
+				return command_args
+		if(IC_FORMAT_COLOR)
+			return sanitize_hexcolor(command_args, null)
+		if(IC_FORMAT_NUMBER)
+			if(istext(command_args))
+				return text2num(command_args)
+			if(isnum(command_args))
+				return command_args
+		if(IC_FORMAT_DIR)
+			if(istext(command_args))
+				return text2dir(command_args)
+		if(IC_FORMAT_BOOLEAN)
+			if(istext(command_args))
+				switch(uppertext(command_args))
+					if("TRUE")
+						return TRUE
+					if("FALSE")
+						return FALSE
```

Comment:
So the write setters should really be doing this already, at least if you use the safe call.

Reply:
Left in for now, I'll go through with a fine-tooth comb and make sure all the public vars are sanitizing as necessary and delete this.